packages/react/src/utils.ts

```diff
@@ -0,0 +1,173 @@
+import { GadgetError, gadgetErrorFor, getNonNullableError, InvalidFieldError } from "@gadgetinc/api-client-core";
+import type { CombinedError, OperationResult } from "@urql/core";
+import { GraphQLError } from "graphql";
+import { Operation, OperationContext, UseQueryState } from "urql";
+
+/**
+ * The inner result object returned from a query result
+ **/
+export interface ReadHookState<Data = any, Variables = Record<string, any>> {
+  fetching: boolean;
+  stale: boolean;
+  data?: Data;
+  error?: ErrorWrapper;
+  extensions?: Record<string, any>;
+  operation?: Operation<Data, Variables>;
+}
+
+/**
+ * The return value of a `useGet`, `useFindMany`, `useFindOne` etc hook.
+ * Includes the data result object and a refetch function.
+ **/
+export declare type ReadHookResult<Data = any, Variables = Record<string, any>> = [
+  ReadHookState<Data, Variables>,
+  (opts?: Partial<OperationContext>) => void
+];
+
+/**
+ * The inner result object returned from a mutation result
+ */
+export interface ActionHookState<Data = any, Variables = Record<string, any>> {
+  fetching: boolean;
+  stale: boolean;
+  data?: Data;
+  error?: ErrorWrapper;
+  extensions?: Record<string, any>;
+  operation?: Operation<Data, Variables>;
+}
+
+/**
+ * The return value of a `useAction`, `useGlobalAction`, `useBulkAction` etc hook.
+ * Includes the data result object and a function for running the mutation.
+ **/
+export declare type ActionHookResult<Data = any, Variables = Record<string, any>> = [
+  ActionHookState<Data, Variables>,
+  (variables?: Variables, context?: Partial<OperationContext>) => Promise<OperationResult<Data, Variables>>
+];
+
+const generateErrorMessage = (networkErr?: Error, graphQlErrs?: GraphQLError[]) => {
+  let error = "";
+  if (networkErr !== undefined) {
+    return (error = `[Network] ${networkErr.message}`);
```

Comment:
nit: Did you intend to `return` here? If so, you can drop the `error =`, and perhaps move the `let` below to show it's only needed below.

Also, what if `networkErr` was `undefined`, and `graphQlErrs` was either `undefined` or an empty array? Would the intent be to still return an empty error string, or some default message?